stac/harness/ilp.py

```diff
@@ -0,0 +1,188 @@
+""" ILP decoding """
+from __future__ import print_function
+
+import os
+import re
+import itertools as itr
+from os import path as fp
+import numpy as np
+from tempfile import mkdtemp
+from shutil import rmtree
+from subprocess import call
+
+from attelo.table import UNRELATED
+from attelo.decoding import Decoder
+
+ZPL_TEMPLATE_DIR = fp.join(fp.dirname(__file__), 'ilp')
+
+
+def pos_indexes(dpack):
+    """ Returns indices of EDUs for each pairing """
+    edu_pos = dict((e, i) for i, e in enumerate(dpack.edus))
+    pair_pos = np.vstack((edu_pos[u], edu_pos[v]) for u, v in dpack.pairings)
+
+    return tuple(pair_pos.transpose())
+
+
+def zimpl_dump(dpack, tgt_dir=None, prefix='default', decoded=False):
+    """ Dump classification scores for use in SCIP
+
+    Default behavior is a dump of dpack.attach and dpack.label
+    If decoded is True, dpack.prediction will be converted
+    """
```

Comment:
Could I get you to use the numpy docstring conventions? Particularly useful are "sections" like Parameters and Returns, see example here: http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_numpy.html